pkg/controllers/resources/services/syncer.go

```diff
@@ -63,53 +66,69 @@ func (s *serviceSyncer) Sync(ctx *synccontext.SyncContext, pObj client.Object, v
-			err := ctx.VirtualClient.Update(ctx, newService)
-			if err != nil {
-				return ctrl.Result{}, err
-			}
-		}
+	// check if recreating service is necessary
+	if vService.Spec.ClusterIP != pService.Spec.ClusterIP {
+		vService.Spec.ClusterIPs = nil
+		vService.Spec.ClusterIP = pService.Spec.ClusterIP
+		ctx.Log.Infof("recreating virtual service %s/%s, because cluster ip differs %s != %s", vService.Namespace, vService.Name, pService.Spec.ClusterIP, vService.Spec.ClusterIP)
 
-		// we will requeue anyways
-		return ctrl.Result{Requeue: true}, nil
-	}
-
-	// check if backwards status update is necessary
-	if !equality.Semantic.DeepEqual(vService.Status, pService.Status) {
-		newService := vService.DeepCopy()
-		newService.Status = pService.Status
-		ctx.Log.Infof("update virtual service %s/%s, because status is out of sync", vService.Namespace, vService.Name)
-		translator.PrintChanges(vService, newService, ctx.Log)
-		err := ctx.VirtualClient.Status().Update(ctx, newService)
+		// recreate the new service with the correct cluster ip
+		err := recreateService(ctx, ctx.VirtualClient, vService)
 		if err != nil {
+			ctx.Log.Errorf("error creating virtual service: %s/%s", vService.Namespace, vService.Name)
 			return ctrl.Result{}, err
 		}
 
-		return ctrl.Result{Requeue: true}, nil
+		return ctrl.Result{Requeue: true}, err
 	}
 
-	// forward update
-	newService = s.translateUpdate(ctx, pService, vService)
-	if newService != nil {
-		translator.PrintChanges(pService, newService, ctx.Log)
+	// patch the service
+	patch, err := patcher.NewSyncerPatcher(ctx, pService, vService)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("new syncer patcher: %w", err)
 	}
+	defer func() {
+		if err := patch.Patch(ctx, pService, vService); err != nil {
+			retErr = utilerrors.NewAggregate([]error{retErr, err})
+		}
+		if retErr != nil {
+			s.EventRecorder().Eventf(vObj, "Warning", "SyncError", "Error syncing: %v", retErr)
+		}
+	}()
+
+	// check
+	sourceService, targetService := synccontext.SyncSourceTarget(ctx, pService, vService)
+
+	// check if we need to update
+	targetService.Spec.ExternalIPs = sourceService.Spec.ExternalIPs
+	targetService.Spec.LoadBalancerIP = sourceService.Spec.LoadBalancerIP
+	targetService.Spec.Ports = sourceService.Spec.Ports
+	targetService.Spec.PublishNotReadyAddresses = sourceService.Spec.PublishNotReadyAddresses
+	targetService.Spec.Type = sourceService.Spec.Type
+	targetService.Spec.ExternalName = sourceService.Spec.ExternalName
+	targetService.Spec.ExternalTrafficPolicy = sourceService.Spec.ExternalTrafficPolicy
+	targetService.Spec.SessionAffinity = sourceService.Spec.SessionAffinity
+	targetService.Spec.SessionAffinityConfig = sourceService.Spec.SessionAffinityConfig
+	targetService.Spec.LoadBalancerSourceRanges = sourceService.Spec.LoadBalancerSourceRanges
+	targetService.Spec.HealthCheckNodePort = sourceService.Spec.HealthCheckNodePort
```

Comment:
couldn't this be 
```go
targetService.Spec = sourceService.Spec 
```
and do what we need with the selectors beforehand?

Reply:
There is many things that shouldn't be updated here (ClusterIPs, IPFamily etc), so its not a bad idea to have a whitelist here